components/calendar/src/types.rs

```diff
@@ -217,40 +228,37 @@ pub struct Time {
 
     /// 0-based second.
     pub second: IsoSecond,
+
+    /// Fractional second
+    pub fraction: FractionalSecond,
 }
 
 impl Time {
     /// Do not validate the numeric input for this component.
-    pub const fn new(hour: IsoHour, minute: IsoMinute, second: IsoSecond) -> Self {
+    pub const fn new(
+        hour: IsoHour,
+        minute: IsoMinute,
+        second: IsoSecond,
+        fraction: FractionalSecond,
```

Comment:
Discussion: Naming of these types. What you have is probably fine, but maybe we should consider:

- `NanoSecond`?
- `IsoNanoSecond`?

Reply:
yeah, NanoSecond sounds good

Reply:
Sounds good. Maybe also name the field `nanosecond` or `nano_second`.

Reply:
I'll go with NanoSecond and nanosecond, thank you for the suggestions!